torch/_inductor/scheduler.py

```diff
@@ -1027,10 +1027,21 @@ def will_fusion_create_cycle(self, node1, node2):
         def check(node):
             if isinstance(node, FusedSchedulerNode) and node not in visited:
                 visited.add(node)
-                return bool(combined_names & node.recursive_predecessors) or any(
-                    check(self.name_to_fused_node[n])
-                    for n in node.recursive_predecessors - combined_predecessors
-                )
+                cond0 = bool(combined_names & node.recursive_predecessors)
+
+                if cond0:
+                    return cond0
+
+                names = node.get_names()
+                shortcut = all(n in combined_predecessors for n in names)
```

Comment:
I'm not sure I see why this is equivalent 🤔

Reply:
If all the fused node outputs are in `combined_predecessors`, then all the `recursive_predecessors` of the node will also be in `combined_predecessors` and thus the intersection will be empty. If `node` actually has something else then we go on the old path.

Reply:
Not sure about the general logic, but this line is equivalent to
```suggestion
                shortcut = name.issubset(combined_predecessors)
```